dlt/common/normalizers/json/relational.py

```diff
@@ -261,6 +258,16 @@ def extend_schema(self) -> None:
             }
         )
 
+        # for every table with the write disposition merge, we propagate the root_key
+        for table_name, table in self.schema.tables.items():
```

Comment:
just one thing: this is called only once when schema is created. so if we add any tables that have write disposition "merge" it won't be reflected in the schema.

we could have a mechanism that whenever we add a new table to the schema we call `extend` or `extend_table` again but I think it is overkill?

Reply:
or maybe not, any time we update or add table we should call `extend_table` even here we should call it for all existing tables

Reply:
done